src/Extension/FluentExtension.php

```diff
@@ -458,4 +458,40 @@ protected function getDataQueryLocale(DataQuery $dataQuery = null)
         }
         return null;
     }
+
+    /**
+     * Determine the link to this object given the specified $locale.
+     * Returns null for DataObjects that do not have a 'Link' function.
+     *
+     * @param string $locale
+     * @return string
+     */
+    public function LocaleLink($locale)
+    {
+        // Skip dataobjects that do not have the Link method
+        if (!$this->owner->hasMethod('Link')) {
+            return '';
+        }
+
+        // Return locale root url if unable to view this item in this locale
+        if ($this->owner->hasMethod('canViewInLocale') && !$this->owner->canViewInLocale($locale)) {
+            $localeObj = Locale::getByLocale($locale);
+            return $localeObj ? $localeObj->getBaseURL() : '';
+        }
+
+        // Mock locale and recalculate link
+        return FluentState::singleton()->withState(function ($newState) use ($locale) {
+            $newState->setLocale($locale);
+
+            $link = DataObject::get($this->owner->ClassName)->byID($this->owner->ID)->Link();
+
+            // Prefix with domain if available
```

Comment:
I'll have a look at this logic; We may need to check if we are in domain mode, first.

Alternatively this should probable be baked into the updateLink() extension method.